concourse/templates/sap-test.libsonnet

```diff
@@ -0,0 +1,150 @@
+{
+  saptestjob:: {
+    local tl = self,
+
+    image:: error 'must be set',
+
+    name: 'sap-workload-test-' + self.image,
+
+    plan: [
+    {
+      get: tl.image + '-gcs',
+      passed: [
+        'publish-to-testing-' + tl.image,
+      ],
+      params: {
+        skip_download: 'true',
+      }
+    },
+    { get: 'guest-test-infra' },
+    {
+      task: 'generate-timestamp',
+      file: 'guest-test-infra/concourse/tasks/generate-timestamp.yaml',
+    },
+    {
+      load_var: 'id',
+      file: 'timestamp/timestamp-ms',
+    },
+    {
+      task: 'generate-post-script',
+      config: {
+        platform: 'linux',
+        image_resource: {
+          type: 'registry-image',
+          source: {
+            repository: 'google/cloud-sdk',
+            tag: 'latest',
+          },
+        },
+        inputs: [
+          { name: "guest-test-infra" },
+        ],
+        run: {
+          path: 'sh',
+          args: [
+            '-exc',
+            |||
+            cd guest-test-infra/concourse/workload-tests/SAP/
+            # We want to upload this actual script with the unique id
+            sed -i 's/$1/gcp-guest-test-outputs/g' sap_post_script.sh
+            sed -i 's/$2/((.:id))/g' sap_post_script.sh
```

Comment:
took me a while to figure out what this was trying to do bc you used bash variables as the replacement markers. typically it's best practice to do e.g. `__BUCKET__` and `__RUNID__` instead in the template. i also think this is just about as ugly as a heredoc, but fine to start with.